fastchat/serve/inference.py

```diff
@@ -160,31 +160,47 @@ def generate_stream(
                 skip_special_tokens=True,
                 spaces_between_special_tokens=False,
             )
+
+            def maybeStop(output, stop_str):
+                for i in range(0, len(output)):
+                    if i > len(stop_str):
+                        return False
+                    if stop_str.startswith(output[-i:]):
+                        return True
+                return False
+
             if stop_str:
                 if isinstance(stop_str, str):
                     pos = output.rfind(stop_str, rfind_start)
                     if pos != -1:
                         output = output[:pos]
                         stopped = True
+                    else:
+                        maybeStop = maybeStop(output, stop_str)
```

Comment:
add comments explaining why this is needed, with a simple example.

Reply:
renamed and added comment.
Use this example but enable streaming.
https://js.langchain.com/docs/getting-started/guide-llm#agents-dynamically-run-chains-based-on-user-input